internal/pkg/v2/infrastructure/redis/client.go

```diff
@@ -279,3 +280,29 @@ func (c *Client) AddDevice(d model.Device) (model.Device, errors.EdgeX) {
 
 	return addDevice(conn, d)
 }
+
+// Update an event
+func (c *Client) UpdateEvent(event model.Event) errors.EdgeX {
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	e, err := c.EventById(event.Id)
+	if err != nil {
+		return err
+	}
+
+	mergeErr := mergo.Merge(&event, e)
```

Comment:
Why do we need to merge when we loaded the Event by Id, changed a property passed the event to this func and now just need to replace the old event?

Reply:
Thanks for pointing this out.  My original implementation is based on what V1 API did.  After further consideration, I've revised the codes to only update database records that are associated with `pushed` timestamp.  As Cloud also mentioned in his comment below, the only event update of V2 API is to update the pushed, so I think this new implementation would be better than removing and adding event.